apps/io-func-wallet-solution-backend/src/wallet-instance-request.ts

```diff
@@ -1,19 +1,19 @@
 import * as t from "io-ts";
 
-import * as E from "fp-ts/Either";
+import * as RTE from "fp-ts/ReaderTaskEither";
 import { pipe } from "fp-ts/function";
 import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { NonceEnvironment, deleteNonce } from "./nonce";
 
-export const WalletInstanceRequest = t.type({
+const WalletInstanceRequest = t.type({
   challenge: NonEmptyString,
   keyAttestation: NonEmptyString,
   hardwareKeyTag: NonEmptyString,
 });
 
 export type WalletInstanceRequest = t.TypeOf<typeof WalletInstanceRequest>;
 
-export const verifyWalletInstanceRequest = (
-  walletInstanceRequest: WalletInstanceRequest
-): E.Either<Error, WalletInstanceRequest> =>
-  // TODO: [SIW-964]: Add nonce validation
-  pipe(walletInstanceRequest, E.right);
+export const validateChallenge = (
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion(naming):** If a function is named _validateSomething_ I expect to have no side effects. This one, instead, deletes an item. I suggest something that express this concept, like _consumeNonce_ (it's just an example). 
Also, I advocate to comment the public function with a docblock‏

Reply:
Since we are talking about **nonce** here and not **challenges** (the challenge is something that the app sends but which could contain something other than the nonce) I would say to rename it to `validateNonce` or `consumeNonce`  as @balanza suggested